veloxms-core/src/main/scala/edu/berkeley/veloxms/models/Model.scala

```diff
@@ -72,30 +78,62 @@ abstract class Model[T: ClassTag] extends Logging {
    * @param nextVersion
    * @return
    */
-  def retrainFeatureModelsInSpark(observations: RDD[(UserID, T, Double)], nextVersion: Version): RDD[(T, FeatureVector)]
+  def retrainFeatureModelsInSpark(
+      observations: RDD[(UserID, T, Double)],
+      nextVersion: Version): RDD[(T, FeatureVector)]
+
+  final def retrainUserWeightsInSpark(
+      itemFeatures: RDD[(T, FeatureVector)],
+      observations: RDD[(UserID, T, Double)],
+      nextVersion: Version)
+    : RDD[(UserID, WeightVector)] = {
+
+    val newAvgUser = retrainAvgUserInSpark(itemFeatures, observations)
+    averageUser.put(newAvgUser, nextVersion)
 
-  final def retrainUserWeightsInSpark(itemFeatures: RDD[(T, FeatureVector)], observations: RDD[(UserID, T, Double)]): RDD[(UserID, WeightVector)] = {
     val obs = observations.map(x => (x._2, (x._1, x._3)))
     val ratings = itemFeatures.join(obs).map(x => (x._2._2._1, x._2._1, x._2._2._2))
     UserWeightUpdateMethods.calculateMultiUserWeights(ratings, numFeatures)
   }
 
   /**
-   * Velox implemented - fetch from local Tachyon partition
+   * Calculates the optimal cold-start user model for the case when we have
+   * no training data for a user.
+   *
+   * Implementation: Take the average score for each item across the
+   * entire training data set and use that as the training data for
+   * the average user model.
```

Comment:
Agreed. It's just a start, and there is probably a more intelligent policy we could use. I actually think the warm-start strategy might be application or model-specific in some cases. But I don't think the system is quite ready to support that yet. If you have any ideas of something else to do that would work for all models, definitely let me know.